packages/aws-library/src/aws_library/ec2/client.py

```diff
@@ -274,6 +275,24 @@ async def get_instances(
         )
         return all_instances
 
+    async def stop_instances(self, instance_datas: Iterable[EC2InstanceData]) -> None:
+        try:
+            with log_context(
+                _logger,
+                logging.INFO,
+                msg=f"stopping instances {[i.id for i in instance_datas]}",
+            ):
+                await self.client.stop_instances(
+                    InstanceIds=[i.id for i in instance_datas]
+                )
+        except botocore.exceptions.ClientError as exc:
```

Comment:
just a thought: what if only the stopping of *some* instances fails, can you handle this without getting into a weird state? ;)

Reply:
this is only the interface code, whatever happens with these instances is the responsibility of the client code. AKA autoscaling service